examples/log.c

```diff
@@ -477,3 +490,35 @@ static int parse_options(
 	return args.pos;
 }
 
+char *find_str(const char *haystack, const char *needle, const int ignore_case) {
+	if(ignore_case)
+		return str_ignorecase_str(haystack, needle);
+	else
+		return strstr(haystack, needle);
+}
+
+char *str_ignorecase_str(const char *haystack_start, const char *needle_start) {
+	char *haystack, *needle;
+	haystack = tolowerstr(haystack_start);
+	needle = tolowerstr(needle_start);
+
+	return strstr(haystack, needle);
+}
+
+char *tolowerstr(const char *str) {
+	char *final;
+	int len = 0;
+	int i = 0;
+	while(str[i++] != '\0')
+		;
+	len = i;
+
+	final = (char *) malloc(sizeof(char) * len);
```

Comment:
Oh! I wasn't aware about the malloc thing and sizeof(char). I was under the impression that the size of char is dependent on the compiler. But I just found that its defined in the standard and its always 1. Thanks!